pkg/apis/certificates/validation/validation.go

```diff
@@ -50,59 +52,53 @@ func ValidateCertificateRequestName(name string, prefix bool) []string {
 func ValidateCertificateSigningRequest(csr *certificates.CertificateSigningRequest) field.ErrorList {
 	isNamespaced := false
 	allErrs := apivalidation.ValidateObjectMeta(&csr.ObjectMeta, isNamespaced, ValidateCertificateRequestName, field.NewPath("metadata"))
-	err := validateCSR(csr)
 
 	specPath := field.NewPath("spec")
-
+	err := validateCSR(csr)
 	if err != nil {
 		allErrs = append(allErrs, field.Invalid(specPath.Child("request"), csr.Spec.Request, fmt.Sprintf("%v", err)))
 	}
 	if len(csr.Spec.Usages) == 0 {
 		allErrs = append(allErrs, field.Required(specPath.Child("usages"), "usages must be provided"))
 	}
-	if csr.Spec.SignerName == nil {
+	if csr.Spec.SignerName == "" {
 		allErrs = append(allErrs, field.Required(specPath.Child("signerName"), "signerName must be provided"))
 	} else {
-		// ensure signerName is of the form domain.com/something and up to 571 characters.
+		// ensure signerName is of the form domain.com/something and up to 635 characters.
 		// This length and format is specified to accommodate signerNames like:
-		// <resource-type-name>.<fully-qualified-group-name>/<resource-namespace>.<resource-name>
-		if len(*csr.Spec.SignerName) > 571 {
-			allErrs = append(allErrs, field.TooLong(specPath.Child("signerName"), *csr.Spec.SignerName, 571))
-		}
-		parts := strings.Split(*csr.Spec.SignerName, "/")
-		if len(parts) != 2 {
-			allErrs = append(allErrs, field.Invalid(specPath.Child("signerName"), *csr.Spec.SignerName, "must be a fully qualified domain of the form 'example.com/signer-name'"))
+		// <resource-type-name>.<fully-qualified-group-name>/<resource-namespace>.<resource-name>.
+		// The max length of a resource type name is 63 characters (DNS1035Label max length)
+		// The max length of a group name is 253 characters (DNS1123Subdomain max length)
+		// The max length of a namespace name is 63 characters (DNS1123Label max length)
+		// The max length of a resource name is 253 characters (DNS1123Subdomain max length)
+		// We then add an additional 3 characters to account for the two '.' and one '/'.
+		signerNameMaxLength := 635
+		if len(csr.Spec.SignerName) > signerNameMaxLength {
+			allErrs = append(allErrs, field.TooLong(specPath.Child("signerName"), csr.Spec.SignerName, signerNameMaxLength))
 		} else {
-			domain := parts[0]
```

Comment:
Can we break the signer name validation into its own function (with early returns).  The `if/else` blocks are hard to keep track of.